lib/dialects/abstract/utils.js

```diff
@@ -0,0 +1,36 @@
+
+const _dialects = {};
+
+/**
+ * Register dialect implementation.
+ *
+ * @param {AbstractDialect} [DialectClass] dialect implemetation as subclass of AbstractDialect.
+ */
+function registerDialect( DialectClass ) {
+  _dialects[DialectClass.getDialectName()] = DialectClass;
```

Comment:
I think we should extend AbstractDialect to have a getDialectName function that needs to be overridden?

Reply:
This is a static method which could not be inherit...use class method is a bit strange.

There may be an approach in TS that `class AbstractDialect implements InterfaceWithGetName`. That might work.

Reply:
Ah we are still in js here. My bad :)